### ISSUE TITLE
How are resources allocated to different apps?

### ISSUE THREAD

**Issue by woudsma** [flynn/flynn#4255]
This is more a question than an issue, but hopefully someone can clarify this for me. (i've read the docs and watched all youtube videos about flynn - not that many - but this is still unclear to me, or i've missed it)  
I'd like to know how `flynn scale` distributes apps across nodes.

#### Some cheap cloud nodes:
- 3 x Ubuntu 16.04, 512mb RAM, 20gb disk

#### A few projects, containing multiple apps:
- project-1
  - app-staging
  - app-production
- project-2
  - frontend-staging
  - frontend-production
  - backend-staging
  - backend-production
- project-3
  - frontend-staging
  - frontend-production
  - backend-staging
  - backend-production
  - worker-staging
  - worker-production
- resources
  - project-2-postgres-staging
  - project-2-postgres-production
  - project-3-postgres-staging
  - project-3-postgres-production
  - file-based-database-container

### Questions:
1. When all processes are scaled `web=1`, are they distributed/replicated across all nodes?
2. What happens when project-2 frontend/backend production apps are scaled to `web=3`, all other apps to `web=2`?
3. What happens when file-based-database-container takes up all disk space on a node? Do all nodes run out of disk space?

Basically, how are (disk space consuming/low priority/high priority) apps distributed between nodes with limited resources?

**Comment by lmars** [comment]
@woudsma 

Jobs are distributed in a round-robin fashion across the cluster on a per app, per process type basis. This means a scale of `web=3` for a given app in a 3 node cluster will lead to 1 `web` job per node, whereas `web=9` would lead to 3 jobs per node.

To answer your questions:

> When all processes are scaled web=1, are they distributed/replicated across all nodes?

Scaling a process to `web=1` means there will be a single `web` job running in the cluster for that app. It is not distributed/replicated because there is only one of them.

> What happens when project-2 frontend/backend production apps are scaled to web=3, all other apps to web=2?

Perhaps I am missing what is being asked here, but there will be 3 `web` jobs running for the first app, and 2 for all the others?

> What happens when file-based-database-container takes up all disk space on a node? Do all nodes run out of disk space?

Then that node runs out of disk space. The other nodes will only run out of disk space if they also have jobs running on them consuming disk space.

> how are (disk space consuming/low priority/high priority) apps distributed between nodes with limited resources?

Flynn does not currently schedule jobs based on available resources, they are just distributed using a round-robin process.
